project_task_notify_state_changed/models/project.py

```diff
@@ -0,0 +1,24 @@
+from odoo import api, fields, models
+
+
+class Task(models.Model):
+    _inherit = 'project.task'
+
+    notify_stage_changed = fields.Boolean(
+        default=True,
+        track_visibility='onchange',
+    )
+
+    @api.multi
+    def _track_template(self, tracking):
+        self.ensure_one()
+        res = super()._track_template(tracking)
+        if not self.notify_stage_changed:
+            res.pop('stage_id', None)
+        return res
```

Comment:
Haven't tested it functionally, but are you sure it's working?
I would've assumed you need to `pop` the `stage_id` before calling super :thinking: